airbyte-cron/src/main/java/io/airbyte/cron/config/DatabaseBeanFactory.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cron.config;
+
+import io.airbyte.config.persistence.ConfigPersistence;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.config.persistence.DatabaseConfigPersistence;
+import io.airbyte.config.persistence.split_secrets.JsonSecretsProcessor;
+import io.airbyte.db.Database;
+import io.airbyte.db.check.DatabaseMigrationCheck;
+import io.airbyte.db.factory.DatabaseCheckFactory;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Value;
+import io.micronaut.flyway.FlywayConfigurationProperties;
+import java.io.IOException;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.sql.DataSource;
+import lombok.extern.slf4j.Slf4j;
+import org.flywaydb.core.Flyway;
+import org.jooq.DSLContext;
+
+/**
+ * Micronaut bean factory for database-related singletons.
+ */
+@Factory
+@Slf4j
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public class DatabaseBeanFactory {
```

Comment:
cc @jdpgrailsdev

Reply:
@pedroslopez Yes...if we move any class that is annotated with Micronaut annotations to a library AND include the proper dependencies so that the annotation processor can find those classes during build time, the class will be automatically picked up by Micronaut at runtime from a library.  I have done this in my airbyte-workers PR with the :airbyte-config:config-models module.  I would also recommend doing this for the `ApplyDefintionsHelper`, which can just have the annotations placed directly on it (e.g. `@Singleton`).